core/src/main/java/org/apache/calcite/plan/RelOptRuleOperand.java

```diff
@@ -104,6 +105,10 @@
       RelOptRuleOperandChildPolicy childPolicy,
       ImmutableList<RelOptRuleOperand> children) {
     assert clazz != null;
+    if (clazz == RelSubset.class) {
+      // Only supports matching subset for leaf operand
+      assert children.size() == 0;
+    }
```

Comment:
Perhaps we don't need this, rule operand should be HepPlanner/VolcanoPlanner agnostic?

Reply:
good point, removed